spec/models/contacts_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe Contact, type: :model do
+  fixtures :contacts
+  fixtures :users
+
+  context 'name is not empty' do
+    it 'return a count of 3 after saving a new entry in the contact table' do
+      Contact.create! name: "Contact3", user_id: 1
+      expect(Contact.all.count).to eq(3)
```

Comment:
This test is not correct, it's only testing that the database has 3 record and not that your action did anything to create this new record. If I update the fixture file I'll have to also update this test. And suppose I update the fixture file so it already has 3 records and also include a different validation that prevents this record creation. You test would still pass. E.g.:
```yml
# ...
three:
  id: 3
  user_id: 1
  name: Contact3
  birthday: 2022-01-01
```
```rb
class Contact
  # ...
  validates :birthday, presence: true # added this new validation
  # ...
end
```

Reply:
Better spec that actually tests what you want here:
```rb
it 'is valid' do
  contact = Contact.new(name: 'John Snow', user: users(:one)
  expect(contact).to be_valid
end
```

Reply:
If you really want to confirm that a new record was added to the database you can accomplish this with
```rb
it 'persists the record to the database' do
  expect { Contact.new(name: 'John Snow', user: users(:one) }
    .to change(Contact, :count)
    .by(1)
end
```